sync-core/src/main/java/com/cloudant/sync/replication/Replication.java

```diff
@@ -41,6 +43,12 @@
 @Deprecated
 public abstract class Replication {
 
+    final List<HttpConnectionRequestFilter> requestFilters = new ArrayList
+            <HttpConnectionRequestFilter>();
+    final List<HttpConnectionResponseFilter> responseFilters = new ArrayList
+            <HttpConnectionResponseFilter>();
```

Comment:
I'd be tempted to make all these public request/responseFilter variables (in all the various classes) private, and then just have `addRequestFilters`, `addResponseFilters` methods in each class. This is breaking encapsulation, to my mind.